src/expr/core/src/expr/expr_udf.rs

```diff
@@ -39,12 +39,16 @@ pub struct UdfExpression {
     client: Arc<ArrowFlightUdfClient>,
     identifier: String,
     span: await_tree::Span,
-    /// Whether a connection error has occurred in the last call.
-    /// If true, we will not retry subsequent calls to prevent blocking the stream.
+    /// Number of remaining successful calls until retry is enabled.
+    /// If non-zero, we will not retry on connection errors to prevent blocking the stream.
+    /// On each connection error, the count will be reset to `INITIAL_RETRY_COUNT`.
+    /// On each successful call, the count will be decreased by 1.
     /// See <https://github.com/risingwavelabs/risingwave/issues/13791>.
-    failing: AtomicBool,
+    disable_retry_count: AtomicU8,
```

Comment:
I can't think of a proper name for this variable. Any suggestions are welcome.

Reply:
What about using the opposite semantics? For example, `continuous_success`.

Reply:
Considering retry enabled is a normal state and retry disabled is a special state, I prefer using the default value 0 to represent the normal state. 🤔